File: 백준/Silver/11723. 집합/집합.py
import sys

N = int(input())

S = set()

for _ in range(N):
    method_list = list(sys.stdin.readline().split())
    method = method_list[0]
    if method == "add":
        S.add(int(method_list[1]))
    elif method == "remove":
        S.discard(int(method_list[1]))
    elif method == "check":
        if int(method_list[1]) in S:
            print(1)
        else:
            print(0)
    elif method == "toggle":
        if int(method_list[1]) in S:
            S.discard(int(method_list[1]))
        else:
            S.add(int(method_list[1]))
    elif method == "all":
        S = set([i+1 for i in range(20)])
    elif method == "empty":
        S = set()

File: 백준/Silver/11723. 집합/집합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int M = Integer.parseInt(br.readLine());
        String command;
        int inputInt;
        int set = 0;
        for (int i = 0; i < M; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            command = st.nextToken();
            switch (command) {
                case "add":
                    inputInt = Integer.parseInt(st.nextToken());
                    set = set | (1<<inputInt);
                    break;
                case "remove":
                    inputInt = Integer.parseInt(st.nextToken());
                    set = set & ~(1<<inputInt);
                    break;
                case "check":
                    inputInt = Integer.parseInt(st.nextToken());
                    if ((set & (1<<inputInt)) != 0){
                        sb.append("1\n");
                    } else {
                        sb.append("0\n");
                    }
                    break;
                case "toggle":
                    inputInt = Integer.parseInt(st.nextToken());
                    if ((set & (1<<inputInt)) == 0){
                        set = set |  1<<inputInt;
                    } else  {
                        set = set & ~(1<<inputInt);
                    }
                    break;
                case "all":
                    set = (1<<21) -1;
                    break;
                case "empty":
                    set = 0;
                    break;
            }
        }
        System.out.println(sb.toString());
    }
}